app/controllers/ems_cloud_controller.rb

```diff
@@ -36,4 +228,72 @@ def set_session_data
   end
+
+  def show_link(ems, options = {})
+    ems_cloud_path(ems.id, options)
+  end
+
+  def set_ems_record_vars(ems, mode = nil)
+    ems.name            = params[:name]
+    ems.provider_region = params[:provider_region]
+    ems.hostname        = params[:hostname]
+    ems.port            = params[:api_port]
+    ems.provider_id     = params[:provider_id]
+    ems.zone            = Zone.find_by_name(params[:zone])
+
+    if ems.kind_of?(ManageIQ::Providers::Microsoft::InfraManager)
+      ems.security_protocol = params[:security_protocol]
+      ems.realm = params[:realm]
+    end
+
+    if ems.kind_of?(ManageIQ::Providers::Vmware::InfraManager)
+      ems.host_default_vnc_port_start = params[:host_default_vnc_port_start].blank? ? nil : params[:host_default_vnc_port_start].to_i
+      ems.host_default_vnc_port_end = params[:host_default_vnc_port_end].blank? ? nil : params[:host_default_vnc_port_end].to_i
+    end
+
+    ems.azure_tenant_id = params[:azure_tenant_id] if ems.kind_of?(ManageIQ::Providers::Azure::CloudManager)
+
+    creds = {}
+    creds[:default] = {:userid   => params[:default_userid],
+                       :password => params[:default_password]} unless params[:default_userid].blank?
+    if ems.supports_authentication?(:metrics) && !params[:metrics_userid].blank?
+      creds[:metrics] = {:userid => params[:metrics_userid], :password => params[:metrics_password]}
+    end
+    if ems.supports_authentication?(:amqp) && !params[:amqp_userid].blank?
+      creds[:amqp] = {:userid => params[:amqp_userid], :password => params[:amqp_password]}
+    end
+    if ems.supports_authentication?(:ssh_keypair) && !params[:ssh_keypair_userid].blank?
+      creds[:ssh_keypair] = {:userid => params[:ssh_keypair_userid], :auth_key => params[:ssh_keypair_password]}
+    end
+    if ems.supports_authentication?(:bearer) && !params[:bearer_token].blank?
+      creds[:bearer] = {:auth_key => params[:bearer_token], :userid => "_"} # Must have userid
+    end
+
+    ems.update_authentication(creds, :save => (mode != :validate))
+  end
+
+  def construct_edit_for_audit(ems)
+    @edit ||= {}
+    ems.kind_of?(ManageIQ::Providers::Azure::CloudManager) ? azure_tenant_id = ems.azure_tenant_id : azure_tenant_id = nil;
+    @edit[:current] = {:name            => ems.name,
+                       :provider_region => ems.provider_region,
+                       :hostname        => ems.hostname,
+                       :azure_tenant_id => azure_tenant_id,
+                       :port            => ems.port,
+                       :provider_id     => ems.provider_id,
+                       :zone            => ems.zone
+    }
+    @edit[:new] = {:name            => params[:name],
+                   :provider_region => params[:provider_region],
+                   :hostname        => params[:hostname],
+                   :azure_tenant_id => params[:tenant_id],
+                   :port            => params[:port],
```

Comment:
@AparnaKarve - curious why we are referencing params[:tenant_id] and not params[:azure_tenant_id]?